### DISCUSSION TITLE
Use BL Touch Needle for Live Bed Adjustment

### DISCUSSION THREAD

**Discussion by BlueWings172** [Jyers/Marlin#1757] [Ideas]
Hello

I don't know if this is possible, but it would be helpful to take advantage of the BL Touch to display live measurements of the bed while the user is rotating the knob to level the bed.

Currently, when using BL Touch to manually level the bed, the BL Touch probes each of the 4 corners and the center. After each probe, it displays a number showing how high or low that corner is, then you have to make approximate adjustment using the knob and probe again and most of the time, you turn the knob too much or too little and you don't get the desired change from first time.

What I am proposing, and I don't know if this is technically possible,  is that the BL Touch needle remains in contact with the bed and while the user is turning the knob, it shows live measurements indicating how close user is from reaching the desired number. 

This way if BL Touch tells you that bottom left corner is 0.2mm low, you turn the knob and see the number getting close to zero at which point you know you have finished from that corner and can move on to the next corner.

thanks

**Comment by BlueWings172** [comment]
@tititopher68-dev 

it is understood that you normally need more than one cycle to accurately level the bed. The idea is to probe the same corner the least times per cycle.

For example, if you probe a corner and it says it is 0.2 low, then you turn the knob left a bit, you probe again and it might say 0.1 low or 0.1 high and then you adjust, probe again and so on until you get to 0.01 or less. But when you have live measurements you stop turning the knob when you are close enough and move on to next corner and do another cycle you're done.

it is like using the BL touch as an electronic digital gauge (https://www.amazon.com/gp/product/B01G9K8U5G/ref=ox_sc_saved_title_8?smid=A1JAAL829ACROK&th=1), if that's even possible.

**Reply by SikVac** [reply]
> @tititopher68-dev
> 
> it is understood that you normally need more than one cycle to accurately level the bed. The idea is to probe the same corner the least times per cycle.
> 
> For example, if you probe a corner and it says it is 0.2 low, then you turn the knob left a bit, you probe again and it might say 0.1 low or 0.1 high and then you adjust, probe again and so on until you get to 0.01 or less. But when you have live measurements you stop turning the knob when you are close enough and move on to next corner and do another cycle you're done.
> 
> it is like using the BL touch as an electronic digital gauge (https://www.amazon.com/gp/product/B01G9K8U5G/ref=ox_sc_saved_title_8?smid=A1JAAL829ACROK&th=1), if that's even possible.

My opinion is that BLTouch is not a linear meter.
Only when it changes state (going from blue to red) is a pulse sent to the processor to display the current modified Z-axis position.
Therefore, a continuous display of the bed position is probably not possible.

**Comment by Cal-LF** [comment]
The probe is not a measuring device, only on and off.

**Comment by BlueWings172** [comment]
Is something like this possible ? They also use a BL touch.

https://www.youtube.com/watch?v=iXtuS8pXz94&start=190 

This looks much faster as the firmware continuously probes the bed while you are turning the knob and will tell you when you are close enough (notice the indication on the screen). this would save at least half of the time.

**Comment by BlueWings172** [comment]
@tititopher68-dev 

it is certainly BL touch. Watch at 2:50

I don't think there is anything proprietary here. the firmware asks the BL Touch to continue probing the bed and displays the reading while you are turning the knob.

For the record, I'm only talking about what is demonstrated between 3:11 and 3:26 and it is basically BL touch continue probing and showing the reading while you turn the knob and stop at desired level.

**Reply by BlueWings172** [reply]
Jyers firmware already has manual bed leveling with bltouch. it is far far better than using a sheet of paper.

the only difference is that currently with Jyers firmware you have to keep pressing the screen button to probe again and again while in that video the probe continuously probes witch is faster. it's no big deal, it is great the way it is now. I just thought a bit less clicking might be better.

**Reply by BlueWings172** [reply]
Sorry maybe I wasn't clear. I'm not talking about UBL mesh tilting or anything related to UBL. On my printer running Jyers firmware, the BL Touch is used during manual bed leveling (once you check the box Use Probe)  and it gives a reading how high or low a corner is in order to turn the knob accordingly. this is different than manually adjusting the UBL mesh points.


![20220106_183706](https://user-images.githubusercontent.com/65966061/148408773-30e17fcb-b884-4c32-aaf4-ac52ce578138.jpg)

**Reply by BlueWings172** [reply]
@tititopher68-dev 

Many thanks brother. will test and let you know as soon I find some energy to re-assemble my printer (I might have contracted COVID despite 3 shots + precautions) and I have disassembled most of the frame just a day before.

**Reply by BlueWings172** [reply]
I have tested the firmware and it doesn't seem to work on my printer. Once you activate Use Probe and then Continuous Probing, then select the corner, the printer probes the corner twice then stops and you cannot exit, cancel or do anything else other than turn off the printer.


![firwmare test](https://user-images.githubusercontent.com/65966061/148983071-de868c15-c399-44b7-b3b7-a1bf645f1966.jpg)

**Reply by BlueWings172** [reply]
where can i find the new fixed version? The link above seem to point to old files.
